kvx/src/implementations/disk.rs

```diff
@@ -147,6 +152,38 @@ impl WriteStore for Disk {
 
         Ok(())
     }
+
+    fn migrate_namespace(&mut self, namespace: kvx_types::NamespaceBuf) -> Result<()> {
+        let root_parent = self.root.parent().ok_or(Error::NamespaceMigration(format!(
```

Comment:
There is no IO error at this stage. This can only fail if the `root` is a directory without a parent. This is actually impossible... as `root` would then have to be `/`. Instead `root` is always something specific under the base storage URL dir _and_ the namespace dir under that. Theoretically, we could also `unwrap` here, but I prefer something a little more readable.

Reply:
Oh, right. `Path::parent` returns an option. My bad.